lib/puppet/application_support.rb

```diff
@@ -14,21 +14,24 @@ module ApplicationSupport
     # before being set in a pushed Puppet Context.
     #
     # @param run_mode [Puppet::Util::RunMode] Puppet's current Run Mode.
+    # @param environment_mode [Puppet::Util::EnvironmentMode] Puppet's current Environment Mode.
     # @return [void]
     # @api private
-    def self.push_application_context(run_mode)
+    def self.push_application_context(run_mode, environment_mode)
       Puppet.push_context(Puppet.base_context(Puppet.settings), "Update for application settings (#{run_mode})")
       # This use of configured environment is correct, this is used to establish
       # the defaults for an application that does not override, or where an override
       # has not been made from the command line.
       #
       configured_environment_name = Puppet[:environment]
-      if run_mode.name == :agent
+      if run_mode.name == :agent || environment_mode.name == :remote
```

Comment:
This is interesting development, but not sure where we'd want to use it yet. We could probably drop it until we have a need.

Reply:
I still approve of explicit symbols for all options rather than a bool, makes it easier to add it back later.